mapie/time_series_regression.py

```diff
@@ -0,0 +1,312 @@
+        # (n_samples_test, n_estimators_). The method
+        # ``aggregate_with_mask`` fits it to the right size
+        # thanks to the shape of k_.
+
+        y_pred_multi = self.aggregate_with_mask(y_pred_multi, self.k_)
+        return y_pred_multi
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        beta_optimize: bool = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Correspond to 'Conformal prediction for dynamic time-series'.
+
+        Parameters
+        ----------
+        beta_optimize: bool
+            Whether to optimize the PIs' width or not.
+
+        """
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
+        X = check_array(X, force_all_finite=False, dtype=["float64", "object"])
+        y_pred = self.single_estimator_.predict(X)
+
+        if alpha is None:
+            return np.array(y_pred)
+        else:
+            alpha_np = cast(NDArray, alpha)
+            check_alpha_and_n_samples(alpha_np, len(self.conformity_scores_))
+
+            if beta_optimize:
+                betas_0 = self._beta_optimize(
+                    alpha=alpha_np,
+                    lower_bounds=self.conformity_scores_.reshape(1, -1),
+                    upper_bounds=self.conformity_scores_.reshape(1, -1),
+                )
+            else:
+                betas_0 = np.full(
+                    shape=(len(alpha), len(self.conformity_scores_)),
+                    fill_value=np.nan,
+                    dtype=float,
+                )
+                for ind_alpha, _alpha in enumerate(alpha):
+                    betas_0[ind_alpha, :] = _alpha / 2.0
+
+            lower_quantiles = np_nanquantile(
+                self.conformity_scores_,
+                betas_0[:, 0],
+                axis=0,
+                method="lower",
+            ).T  # type: ignore
+            higher_quantiles = np_nanquantile(
+                self.conformity_scores_,
+                1 - alpha_np + betas_0[:, 0],
+                axis=0,
+                method="higher",
+            ).T  # type: ignore
+
+            if self.cv == "prefit":
+                y_pred_low = y_pred[:, np.newaxis] + lower_quantiles
+                y_pred_up = y_pred[:, np.newaxis] + higher_quantiles
+            else:
+                y_pred_multi = self._pred_multi(X)
+                pred = aggregate_all(self.agg_function, y_pred_multi)
+                lower_bounds, upper_bounds = pred, pred
+
+                y_pred_low = np.column_stack(
+                    [
+                        lower_bounds + lower_quantiles[k]
+                        for k, _ in enumerate(alpha_np)
+                    ]
+                )
+                y_pred_up = np.column_stack(
```

Comment:
Can be simplified to `y_pred_up = lower_bounds + higher_quantiles.reshape(-1, 1)` ?

Reply:
Indeed